Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5668,6 +5671,8 @@ void RenderLayer::updateSelfPaintingLayer()
         return;
 
     m_isSelfPaintingLayer = isSelfPaintingLayer;
+    setNeedsPositionUpdate();
```

Comment:
Does that imply `isVisibilityHiddenOrOpacityZero` status changed? I did not entirely understand why you need this here.